File: tests/fixtures/supabase/supabase-order-by-timestamp-not-identity-fixed/02-two-chains-adversarial.ts
import { getSupabase } from '../../../lib/supabase.js';

// Adversarial: two separate query chains close together in the same file.
// Chain A orders by "id" but selects no timestamp column at all, so the
// rule must not flag it. Chain B selects created_at but already orders by
// created_at itself. A naive file-level (rather than per-chain) tracker
// could wrongly let chain B's timestamp column "leak" into chain A's
// order-by-id check — this fixture proves chain isolation holds.
export async function listLogsAndEvents() {
  const logs = await getSupabase().from('logs').select('id, level').order('id', { ascending: false });

  const events = await getSupabase()
    .from('events')
    .select('id, created_at')
    .order('created_at', { ascending: false });

  return { logs: logs.data, events: events.data };
}

File: tests/fixtures/supabase/supabase-order-by-timestamp-not-identity-fixed/02_two_chains_adversarial.py
from lib.client import supabase


def list_plain_ids():
    # No timestamp column selected — ordering by "id" is not a smell here.
    return supabase.table("tags").select("id, name").order("id").execute()


def list_history_two_calls():
    query = supabase.table("history").select("id, created_at")
    # Chained onto a variable, not a literal CallExpression — chain-state
    # tracking can't link back to the `.select()` here (same limitation as
    # the JS rule's chainObjectCall), so this is a known blind spot rather
    # than a validated safe case.
    return query.order("id").execute()
